src/openrct2/management/finance.c

```diff
@@ -125,17 +122,15 @@ void finance_pay_research()
  */
 void finance_pay_interest()
 {
-    money32 current_loan = gBankLoan;
-    uint8 current_interest = gBankLoanInterestRate;
-
-    // This variable uses the 64-bit type as the line below can involve multiplying very large numbers
-    // that will overflow money32 (e.g. in the Alton Towers RCT1 scenario)
-    money64 tempcost = (current_loan * 5 * current_interest) >> 14; // (5 * interest) / 2^14 is pretty close to
-
     if (gParkFlags & PARK_FLAGS_NO_MONEY)
         return;
 
-    finance_payment(tempcost, RCT_EXPENDITURE_TYPE_INTEREST);
+    const money32 loanMultiplicator = 5 * gBankLoanInterestRate;
+    money32 currentLoanLow = (loanMultiplicator * (gBankLoan & 0x3FFF)) >> 14; // multiply first, then bitshift.
```

Comment:
0x3FFF is a magic number you introduced without any explanation.

Also, Coding Style says comments should go above a line, not next to it.